pandas/tests/test_series.py

```diff
@@ -5511,6 +5511,17 @@ def test_astype_str(self):
                 expec = s.map(compat.text_type)
                 assert_series_equal(res, expec)
 
+        # GH9757
```

Comment:
could you add a test for converting a dataframe, ideally one with multiple dtypes?

Also, this would test for above `astype(unicode)` and `astype(str)` on Python 2.